e2e/scripts/smoke_test.sh

```diff
@@ -0,0 +1,119 @@
+#!/bin/bash -e
+
+test_source_repo=$(pwd)
+docker_test_repo=/workspace/go/src/github.com/rook/rook
+git_smoke_test_directory=github.com/rook/rook/e2e/tests/smoke
+container_image=quay.io/quantum/rook-test
+tmp_docker_sock_path=/tmp/docker.sock
+results_dir=results
+results_filename=test.log
+
+#create the rook infrastructure container
+rook_infra::create() {
+    export id=$(docker run \
+        --net=host \
+        -d \
+        -e container=docker \
+        -e GOPATH=/workspace/go \
+        -e GOROOT=/usr/lib/go \
+        --privileged \
+        --security-opt=seccomp:unconfined \
+        --cap-add=SYS_ADMIN \
+        -v /dev:/dev \
+        -v /sys/fs/cgroup:/sys/fs/cgroup \
+        -v /sbin/modprobe:/sbin/modprobe \
+        -v /lib/modules:/lib/modules:rw \
+        -v /var/run/docker.sock:${tmp_docker_sock_path} \
+        -v ${test_source_repo}:${docker_test_repo} \
+        -w ${docker_test_repo}/e2e \
+        ${container_image} \
+        /sbin/init)
+
+
+        rc=$?; if [[ $rc != 0 ]]; then set -e; fi
+}
+
+#prepare the rook_infra container to run tests
+rook_infra::init() {
+    echo Creating results directory
+    docker exec ${id} mkdir -p ${results_dir}
+    rc=$?; if [[ $rc != 0 ]]; then set -e; fi
+
+    echo Removing infra docker.sock
+    docker exec ${id} rm -rfv /var/run/docker.sock
```

Comment:
im wondering if you still need to do this socket remove/symlink commands. You can do the same thing you did for `rbd`. You can mount the outer docker socket to a different file and pass the correct DOCKER_HOST env when executing docker commands that apply to the outer docker.